115-update-KafkaMirrorMaker2-resource-to-better-correspond-its-use.md

```diff
@@ -0,0 +1,203 @@
+The cluster that will be used by the underlying Connect cluster is specified by its alias in the `.spec.connectCluster` field.
+
+The `.spec.mirrors` array contains the list of the different MirrorMaker connectors.
+Each _mirror_ can specify target (`targetCluster`) and source cluster (`sourceCluster`) by referring to it through the `alias`.
+However, the `targetCluster` has to always be only the same cluster which already serves as the Connect cluster.
+Therefore the `targetCluster` field in the mirror specification has to be the same as the `.spec.Connect` cluster.
+
+Each mirror can configure 3 different connectors:
+* Source connector for mirroring Kafka records
+* Checkpoint connector for mirroring committed offsets
+* Heartbeat connector
+
+As explained in [strimzi/strimzi-kafka-operator#11842](https://github.com/strimzi/strimzi-kafka-operator/issues/11842), the heartbeat does not really fit the pattern as it should be used in the opposite direction than the source and checkpoint connectors.
+It is also rarely used by Strimzi users.
+
+## Motivation
+
+The current structure of the `KafkaMirrorMaker2` resources does not correspond to the current way it is used.
+* The way the clusters are configured (as a Connect, source, and target clusters) makes it hard to validate in the CRD validation schema.
+  We often can validate these things only in the operator.
+  That often means users miss the errors as they are visible in the logs and the `.status` section of the custom resource but not as a result of the `kubectl` commands to create or change the resource.
+* The `targetCluster` field in the mirror configuration has to be always the same value as the `.spec.connectCluster` and is not really _configurable_.
+* It makes it hard to enforce and validate the configuration of the internal Connect topics and consumer group ID.
+  See the [_Enforce the configuration of the internal Kafka Connect names and `group.id`_ proposal](https://github.com/strimzi/proposals/pull/176) for more details.
+* The heartbeat connector configuration does not make much sense in the relationship to the source and target clusters.
+  See [strimzi/strimzi-kafka-operator#11842](https://github.com/strimzi/strimzi-kafka-operator/issues/11842) for more details.
+
+This proposal aims to improve these things.
+
+## Proposal
+
+This proposal suggests deprecating the following fields and remove them in the `v1` CRD API version:
+* `.spec.connectCluster` field
+* `.spec.clusters` list
+* `heartbeatConnector` section in the mirror configuration (`.spec.mirrors[].heartbeatConnector`)
+* `targetCluster` field in the mirror configuration (`.spec.mirrors[].targetCluster`)
+
+These will be replaced with the following new fields:
+* `.spec.targetCluster` section for configuring the cluster that will be used as target and Connect cluster
+  The new `targetCluster` configuration will differ from the current `.spec.clusters` configuration:
+    * There will be no `alias` field
```

Comment:
The target alias is used when labelling metrics, so I think we should keep that in the configuration, e.g. the MirrorSourceConnector emits metrics like:
```
kafka.connect.mirror:type=MirrorSourceConnector,source=<SOURCE_ALIAS>,
target=<TARGET_ALIAS>,topic=<TOPIC>,partition=<PARTITION>
```